fuse_constraints/test/test_absolute_pose_3d_stamped_constraint.cpp

```diff
@@ -0,0 +1,205 @@
+         0.4, 0.4, 0.1, 0.3, 5.0, 0.5,
+         0.5, 0.3, 0.2, 0.4, 0.5, 6.0;
+
+  auto constraint = AbsolutePose3DStampedConstraint::make_shared(
+    *position_variable,
+    *orientation_variable,
+    mean,
+    cov);
+
+  // Build the problem
+  ceres::Problem problem;
+  problem.AddParameterBlock(
+    position_variable->data(),
+    position_variable->size(),
+    position_variable->localParameterization());
+  problem.AddParameterBlock(
+    orientation_variable->data(),
+    orientation_variable->size(),
+    orientation_variable->localParameterization());
+
+  std::vector<double*> parameter_blocks;
+  parameter_blocks.push_back(position_variable->data());
+  parameter_blocks.push_back(orientation_variable->data());
+
+  problem.AddResidualBlock(
+    constraint->costFunction(),
+    constraint->lossFunction(),
+    parameter_blocks);
+
+  // Run the solver
+  ceres::Solver::Options options;
+  ceres::Solver::Summary summary;
+  ceres::Solve(options, &problem, &summary);
+
+  // Check
+  EXPECT_NEAR(1.0, position_variable->x(), 1.0e-5);
+  EXPECT_NEAR(2.0, position_variable->y(), 1.0e-5);
+  EXPECT_NEAR(3.0, position_variable->z(), 1.0e-5);
+  EXPECT_NEAR(1.0, orientation_variable->w(), 1.0e-3);
+  EXPECT_NEAR(0.0, orientation_variable->x(), 1.0e-3);
+  EXPECT_NEAR(0.0, orientation_variable->y(), 1.0e-3);
+  EXPECT_NEAR(0.0, orientation_variable->z(), 1.0e-3);
+
+  // Compute the covariance
+  std::vector<std::pair<const double*, const double*> > covariance_blocks;
+  covariance_blocks.emplace_back(position_variable->data(), position_variable->data());
+  covariance_blocks.emplace_back(orientation_variable->data(), orientation_variable->data());
+  covariance_blocks.emplace_back(position_variable->data(), orientation_variable->data());
+
+  ceres::Covariance::Options cov_options;
+  ceres::Covariance covariance(cov_options);
+  covariance.Compute(covariance_blocks, &problem);
+  std::vector<double> cov_pos_pos(position_variable->size() * position_variable->size());
+  covariance.GetCovarianceBlock(position_variable->data(), position_variable->data(), cov_pos_pos.data());
+
+  std::vector<double> cov_or_or(orientation_variable->size() * orientation_variable->size());
+  covariance.GetCovarianceBlock(orientation_variable->data(), orientation_variable->data(), cov_or_or.data());
+
+  std::vector<double> cov_pos_or(position_variable->size() * orientation_variable->size());
+  covariance.GetCovarianceBlock(position_variable->data(), orientation_variable->data(), cov_pos_or.data());
+
+  // Assemble the full covariance from the covariance blocks
+  Eigen::Map<Eigen::Matrix3d> pos_pos(cov_pos_pos.data());
+  Eigen::Map<Eigen::Matrix<double, 4, 4>> or_or(cov_or_or.data());
+  Eigen::Map<Eigen::Matrix<double, 3, 4, Eigen::RowMajor>> pos_or(cov_pos_or.data());
+
+  Matrix6d actual_covariance;
+  actual_covariance << pos_pos, pos_or.block<3, 3>(0, 1), pos_or.block<3, 3>(0, 1).transpose(), or_or.block<3, 3>(1, 1);
```

Comment:
This test will fail until this is merged:

https://github.com/locusrobotics/fuse/pull/23